doc/specs/#2697 - PowerShell DSC resources.md

```diff
@@ -0,0 +1,256 @@
+            }
+        }
+    }
+}
+
+$testResult = Invoke-DscResource @resource -Method Test
+if (-not $testResult.InDesiredState)
+{
+    Invoke-DscResource @resource -Method Set
+}
+
+```
+
+### WinGetAdminSettings
+This resource will be in charge of enabling and disabling winget administrator settings and is a wrapper around Enable-WinGetSetting and Disable-WinGetSetting.
+
+#### winget-cli
+Currently, there's no way to expose if the administrator settings are enabled or not. This is a requirement for both Test and Get. We need to expose this information with a new `--export` option in the `winget settings` command. The export option will print a json with the information required.
+
+```
+{
+    "LocalManifestFiles": true,
+    "BypassCertificatePinningForMicrosoftStore": false,
+    "FutureAdminSetting": true
+}
+```
+We could technically read the registry and avoid this, but that doesn't sound like the right thing to do.
+
+#### DSC Resource
+
+```
+WinGetAdminSettings [String] #ResourceName
+{
+    Settings = [string[]]
+    SID = [string]
+    [DependsOn = [string[]]]
+    [PsDscRunAsCredential = [PSCredential]]
+}
+```
+
+*Settings*
+
+A Hashtable with the admin settings to be set.
+
+*SID (Key)*
+
+See SID property of UserSettings
+
+#### Example of usage
+```
+$resource = @{
+    Name = 'WinGetAdminSettings'
+    ModuleName = 'Microsoft.WinGet.DSC'
+    Property = @{
+        Settings = @{
+            LocalManifestFiles = $true,
+        }
+    }
+}
+
+$testResult = Invoke-DscResource @resource -Method Test
+if (-not $testResult.InDesiredState)
+{
+    Invoke-DscResource @resource -Method Set
+}
+
+```
+
+### WinGetSources
+This resource will be in charge of adding or removing winget sources. It wraps calls to Add-WinGetSource, Remove-WinGetSource and Get-WinGetSource.
+
+#### DSC Resource
+```
+WinGetSources [String] #ResourceName
+{
+    SID = [string]
+    Sources = [WinGetSource[]]
+    [DependsOn = [string[]]]
+    [Ensure = [string]]
+    [PsDscRunAsCredential = [PSCredential]]
+}
+```
+
+*SID (Key)*
+
+See SID property of UserSettings
+
+*Sources*
+This is an array of a new PowerShell WinGetSource class. It contains the name, type and argument of the source. Type is optional as it default as "Microsoft.PreIndexed.Package". Think about it as the deserialized form of the result of `winget sources export` without the Identifier and Data properties.
+
+It might seem natural to use the type "Microsoft.Management.Deployment.PackageCatalogReference" returned by Get-WinGetSource, but PackageCatalogReference is not what we really want.
+
+Existing PowerShell DSC property Ensure will tell if the sources need to be added or deleted. Test will verify that all the sources are present.
+
+*Ensure*
+
+This is a enum with two values: Absent and Present. Defaults to present. For Set, Absent means call Remove-WinGetSource and Ensure means if not there call Add-WinGetSource otherwise call Reset-WinGetSource to repair it.
```

Comment:
I think it might be worth having a separate property for reset rather than always defaulting to doing it.